api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -105,15 +105,12 @@ public void setConceptDAO(ConceptDAO dao) {
 	 * @should leave preferred name preferred if set
 	 * @should set default preferred name to fully specified first
 	 * @should not set default preferred name to short or index terms
-         * @should force set flag if set members exist
+	 * @should force set flag if set members exist
 	 */
 	public Concept saveConcept(Concept concept) throws APIException {
-		ConceptMapType defaultConceptMapType = null;
+		ConceptMapType defaultConceptMapType = Context.getConceptService().getDefaultConceptMapType();
 		for (ConceptMap map : concept.getConceptMappings()) {
 			if (map.getConceptMapType() == null) {
-				if (defaultConceptMapType == null) {
-					defaultConceptMapType = Context.getConceptService().getDefaultConceptMapType();
-				}
```

Comment:
@mavrk can you please remove this change. remember my comment that this issue is about adding tests not refactoring. thank you!!